power_grid_model_c/power_grid_model/include/power_grid_model/main_core/update.hpp

```diff
@@ -96,22 +101,97 @@ inline void update_inverse(MainModelState<ComponentContainer> const& state, Forw
+    auto query_param_in_seq_map = [&seq_idx_map](IdxVector const& param_idx_vec) {
+        std::vector<Idx2D> result;
+        for (auto const& param_idd : param_idx_vec) {
+            auto const& seq_idx = (*seq_idx_map)[param_idd];
+            result.insert(result.end(), seq_idx.begin(), seq_idx.end());
+        }
+        return result;
+    };
+
+    auto branch_params = query_param_in_seq_map(y_bus.get_branch_param_idx());
+    auto shunt_params = query_param_in_seq_map(y_bus.get_shunt_param_idx());
+
+    MathModelParamIncrement<sym> math_model_param_incrmt;
+    math_model_param_incrmt.branch_param_to_change.reserve(branch_params.size());
+    math_model_param_incrmt.shunt_param_to_change.reserve(shunt_params.size());
+    std::ranges::transform(branch_params, std::back_inserter(math_model_param_incrmt.branch_param_to_change),
+                           [](const Idx2D& val) { return val.pos; });
+    std::ranges::transform(shunt_params, std::back_inserter(math_model_param_incrmt.shunt_param_to_change),
+                           [](const Idx2D& val) { return val.pos; });
+
+    // check changes in case sequence_idx_map is available but indicates empty
+    // this back-up check fixes the edge case where the sequence_idx_map is available but indicates empty
+    if (branch_params.size() == 0) {
+        auto branch_param_to_change_views =
+            boost::irange(y_bus_param.branch_param.size()) |
+            boost::adaptors::filtered([&math_model_param, &y_bus_param](Idx i) {
+                return cmplx_neq<sym>(math_model_param->branch_param[i].yff(), y_bus_param.branch_param[i].yff()) ||
+                       cmplx_neq<sym>(math_model_param->branch_param[i].yft(), y_bus_param.branch_param[i].yft()) ||
+                       cmplx_neq<sym>(math_model_param->branch_param[i].ytf(), y_bus_param.branch_param[i].ytf()) ||
+                       cmplx_neq<sym>(math_model_param->branch_param[i].ytt(), y_bus_param.branch_param[i].ytt());
+            });
+        math_model_param_incrmt.branch_param_to_change = {branch_param_to_change_views.begin(),
+                                                          branch_param_to_change_views.end()};
+    }
+
+    if (shunt_params.size() == 0) {
+        auto shunt_param_to_change_views =
+            boost::irange(y_bus_param.shunt_param.size()) |
+            boost::adaptors::filtered([&math_model_param, &y_bus_param](Idx i) {
+                return cmplx_neq<sym>(math_model_param->shunt_param[i], y_bus_param.shunt_param[i]);
+            });
+        math_model_param_incrmt.shunt_param_to_change = {shunt_param_to_change_views.begin(),
+                                                         shunt_param_to_change_views.end()};
+    }
+
+    auto param_incrmt_ptr = std::make_shared<MathModelParamIncrement<sym> const>(std::move(math_model_param_incrmt));
+
+    y_bus.update_admittance_increment(math_model_param, param_incrmt_ptr);
+    /* param, changed_param */
```

Comment:
what's the deal with this?

Reply:
Could you be more specific here? I wasn't sure what you were referring to here. Is it: a) why do you call the function here? b) why are the function/variable named the way they are? or c) why do you need a parameter explanation in a comment line?

Reply:
I think c). Resolved in latest commit.

Reply:
c) 😬  even more so because it's an inline comment instead of a regular comment

Reply:
Well, it used to have FOUR parameters and I put them somewhere nice and aligned... that was until the formatter and removal of the two params ofc.

Reply:
Then you can remove it 😛